spark-plugins/src/main/java/co/cask/hydrator/plugin/spark/LogisticRegressionTrainer.java

```diff
@@ -0,0 +1,126 @@
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.hydrator.plugin.spark;
+
+
+import co.cask.cdap.api.annotation.Description;
+import co.cask.cdap.api.annotation.Name;
+import co.cask.cdap.api.annotation.Plugin;
+import co.cask.cdap.api.data.format.StructuredRecord;
+import co.cask.cdap.api.data.schema.Schema;
+import co.cask.cdap.api.dataset.lib.FileSet;
+import co.cask.cdap.etl.api.PipelineConfigurer;
+import co.cask.cdap.etl.api.batch.SparkExecutionPluginContext;
+import co.cask.cdap.etl.api.batch.SparkSink;
+import org.apache.spark.api.java.JavaRDD;
+import org.apache.spark.api.java.JavaSparkContext;
+import org.apache.spark.api.java.function.Function;
+import org.apache.spark.mllib.classification.LogisticRegressionModel;
+import org.apache.spark.mllib.classification.LogisticRegressionWithLBFGS;
+import org.apache.spark.mllib.feature.HashingTF;
+import org.apache.spark.mllib.regression.LabeledPoint;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import javax.annotation.Nullable;
+
+/**
+ * Spark Sink plugin that trains a model based upon various labels in the structured record.
+ * Writes this model to a file of a FileSet.
+ */
+@Plugin(type = SparkSink.PLUGIN_TYPE)
+@Name(LogisticRegressionTrainer.PLUGIN_NAME)
+@Description("Uses Logistic Regression to train a model based upon whether messages are spam or not.")
+public class LogisticRegressionTrainer extends SparkMLTrainer {
+  public static final String PLUGIN_NAME = "LogisticRegressionTrainer";
+
+  private Config config;
+
+  /**
+   * Configuration for the LogisticRegressionTrainer.
+   */
+  public class Config extends MLTrainerConfig {
+
+    @Nullable
+    @Description("The number of features to use in training the model. It must be of type integer and equal to the " +
+      "number of features used in LogisticRegressionClassifier. The default value if none is provided will be " +
+      "100.")
+    private final Integer numFeatures;
+
+    @Nullable
+    @Description("The number of classes to use in training the model. It must be of type integer. " +
+      "The default value if none is provided will be 2.")
+    private final Integer numClasses;
+
+    public Config() {
+      this.numFeatures = 100;
+      this.numClasses = 2;
+    }
+
+    public Config(String fileSetName, String path, @Nullable String featureFieldsToInclude, String labelField,
+                  @Nullable String featureFieldsToExclude, @Nullable Integer numFeatures,
+                  @Nullable Integer numClasses) {
+      super(fileSetName, path, featureFieldsToInclude, featureFieldsToExclude, labelField);
+      this.numFeatures = numFeatures;
+      this.numClasses = numClasses;
+    }
+  }
+
+  @Override
+  public void configurePipeline(PipelineConfigurer pipelineConfigurer) {
+    configure(pipelineConfigurer, config.fileSetName, config.featureFieldsToInclude, config.featureFieldsToExclude,
```

Comment:
Rather than having a method that just calls a parent method, it is better for the parent to just implement the method and for the child to leave it out.

In this case, the parent should implement configurePipeline, and the method should not be implemented in this class.